.github/workflows/lint-python.yml

```diff
@@ -0,0 +1,19 @@
+---
+name: Check python syntax
+
+"on":
+  push:
+    paths:
+      - '**.py'
+      - .github/workflows/lint-python.yml
+
+jobs:
+  lint-python-syntax:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        with:
+          python-version: '3.x'
+      - run: pip3 install flake8
+      - run: flake8
```

Comment:
flake8 is a good baseline. Nothing to disagree with (once E501 is disabled or configured differently) -- except maybe the two empty lines required b/f every function, but this is minor.

I tend to use pylint which goes much further in its advice. But obviously we can not require 10/10 points. Maybe 9/10?

Or maybe we leave flake8 as CI in there and recommend coders to use pylint in addition?

Reply:
Well actually, I would vote for disabling E226 as well (enforcing spaces around operators).
While I'm not against being forced to write ` a = b + 1` instead of `a = b+1`, I might find `a = 2*b +1` better readable than `a = 2 * b + 1`. (And I would certainly want to enforce the spaces around the assignment operator.) More importantly `substr = stg[idx+1:]` is significantly more readable than `substr = stg[idx + 1:]`, which E226 complains about unfortunately.
Obviously, I know this is a question of taste ... which is hard to argue about.

Reply:
Going though more software, I ended up disabling:
```
    E251 unexpected spaces around keyword / parameter equals
    E226 missing whitespace around arithmetic operator
    E221 multiple spaces before operator
    E501 line too long (XXX > 79 characters)
```
E251: Readability can be improved by spaces around keyword arguments (when you have long lists with multiple lines
E221: Aligning = statements in lines below each other can improve readability

Reply:
Maybe I should explain by example:
E251
`
retval = function(kwd1 = value1+1,
                  kwd2 = function2(value2, kwdarg=value3),
                  kwd3 = astring[-4:])
`
is less readable without the spaces around the kwd[123] = assignments, IMVHO.

Reply:
E221:
```
attr1 = obj.attr1
at2   = obj.at2
att3  = obj.att3
```
or
```
val1 = expr1 + 1*val
val2 = ex2  + 2*val
val3 = exp3  + 3*val
```
visually helps to avoid mistakes.